bin/i18n/resources/dashboard/course_content/sync_in.rb

```diff
@@ -200,6 +200,26 @@ def process
                   i18n_strings[BLOCK_CATEGORIES_TYPE][name] = name if name
                 end
 
+                ## Lab2 Function Names
+                if level.uses_lab2?
+                  functions = level.
+                    try(:level_data).
+                    try(:[], "startSources").
+                    try(:[], "blocks").try(:[], "blocks")&.
+                    filter {|block| block.try(:[], "type") == "procedures_defnoreturn"}
+
+                  i18n_strings['function_definitions'] = Hash.new unless functions.nil? || functions.empty?
+
+                  functions&.each do |function|
+                    name = function.try(:[], "fields").try(:[], "NAME")
+                    # The name is used to uniquely identify the function. Skip if there is no name.
+                    next unless name
+                    function_definition = Hash.new
+                    function_definition["name"] = name
+                    i18n_strings['function_definitions'][name] = function_definition
```

Comment:
It looks like `name` is both key and value. If I understand correctly, the final structure of the hash would be:
```
'function_definitions': {
  'fucntion_name': {
    'name': 'fucntion_name'
  }
}
```
Is this correct?

Reply:
I believe so.  Here's a random example in our repo:
https://github.com/code-dot-org/code-dot-org/blob/2d735f3e93b139a9179df9cbd90ef2bf5f996149/i18n/locales/source/course_content/2018/coursee-2018.json#L1595-L1599

And here's an example of what this code generates:
```
    "function_definitions": {
      "bass": {
        "name": "bass"
      }
    }
```